packages/@aws-cdk/core/lib/names.ts

```diff
@@ -36,5 +39,26 @@ export class Names {
     return components.length > 0 ? makeUniqueId(components) : '';
   }
 
+  /**
+   * Returns a CloudFormation-compatible unique identifier for a construct based
+   * on its path. If the node is a Stack, the stack name is used for that path component instead.
+   * The identifier includes a human readable portion rendered from the path components and a hash suffix.
+   * @param construct The construct
+   * @param maxUniqueIdLength The maximum length of the unique id
+   * @returns a unique id based on the construct path
+   */
+  public static stackSafeUniqueId(construct: Construct, maxUniqueIdLength: number = 256): string {
```

Comment:
I actually don't really like this `maxUniqueIdLength` parameter in this API. Can't the caller just truncate this to whatever length they need? I feel like this makes it inconsistent with `nodeUniqueId` for relatively little benefit.

Thoughts about removing `maxUniqueIdLength` from here, and moving this truncating part to the caller?

Reply:
I disagree. The benefit is huge! Every caller will realistically need to do this. Punting it to the consumer is just pushing work on them for no reason.

The fact that `nodeUniqueId` didn't have it is a miss, imo.

Reply:
Then how about fixing that mistake, and adding `maxUniqueIdLength` to `uniqueId()` too?

Because right now, we have duplication in handling the max length (we do it once here, and once in the `Pipeline` class). If we did it in both methods in `Names`, we could extract the max length handling piece to a private method/function, and get rid of this duplication.

Reply:
I'm actually looking at deprecating uniqueId since it's not aaaaaactually unique. So, for now I'm not making changes to it, but if we think that it's not viable to actually deprecate it, I'll update it to be consistent. I'd like to do that in a separate PR, though.

Reply:
From my search, `Names.uniqeId()` is used 62 times in production code, so deprecating it might be a little harsh (although I'm not saying it's not possible - just making you aware of the numbers here 😉).

Because of the high number of usages, I think we won't get rid of it anytime soon, so adding the `maxUniqueIdLength` parameter to it makes a lot of sense (in my opinion).